pkg/conversion/meta.go

```diff
@@ -93,6 +95,12 @@ func UpdateVersionAndKind(baseFields []string, versionField, version, kindField,
 
 	field := v.FieldByName(kindField)
 	if !field.IsValid() {
+		// Types defined in the unversioned package are allowed to not have a
+		// kindField. Clients will have to know what they are based on the
+		// context.
+		if pkg == "unversioned" {
```

Comment:
Sorry, I was still thinking about this one. :) Can you add a comment here, `// TODO: add some type trait here, or some way of indicating whether this feature is allowed on a per-type basis. Using package name is overly broad and a bit hacky.`